h_malloc.c

```diff
@@ -1087,6 +1087,29 @@ COLD static void handle_bugs(void) {
         ro.purge_slabs = false;
         ro.region_quarantine_protect = false;
     }
+
+    char cmdline[256];
+    if (readlink("/proc/self/cmdline", cmdline, sizeof(cmdline)) == -1) {
+        return;
+    }
+
+    const char* broken_apps[] = {
+        /* Diablo Immortal */ "com.blizzard.diablo.immortal",
+        /* Jetpack Joyride */ "com.halfbrick.jetpackjoyride"
+    };
+
+    // Workaround for apps (mostly games) which have memory corruption
+    int i = 0;
+    while(broken_apps[i]) {
```

Comment:
Missing space.

Reply:
This loop is incorrect. You haven't included a null pointer at the end of the array but are depending on reaching a null pointer. You should use a `for` loop instead of this approach:

`for (unsigned i = 0; i < sizeof(broken_apps[0]) / sizeof(broken_apps); i++)`

The alternative would be putting NULL at the end of the array for your approach to be correct.